src/nbl/asset/builtin/cache/ICacheKeyCreator.h

```diff
@@ -0,0 +1,26 @@
+#ifndef __I_CACHE_KEY_CREATOR_H_INCLUDED__
+#define __I_CACHE_KEY_CREATOR_H_INCLUDED__
+
+#include "nbl/asset/IAsset.h"
+
+namespace nbl
+{
+	namespace asset
+	{
+		class ICacheKeyCreator
+		{
+			public:
+				virtual std::string to_string() = 0;
+
+				template<typename Type>
+				_NBL_STATIC_INLINE std::string getHexString(const Type& value)
```

Comment:
not `get` call it `make`

Reply:
also I have a better idea.. make it a variable argument template
```cpp
template<typename... T>
static inline std::string getHexString(const T&... values)
{
   std::stringstream stream;
   stream << std::setfill('0') << std::hex;
   stream << ... << args;
   return stream.str();
}
```

so then instead of
`ICacheKeyCreator::getHexString(binding) + ICacheKeyCreator::getHexString(format) + ICacheKeyCreator::getHexString(relativeOffset)`
you can do
`ICacheKeyCreator::getHexString(binding,format,relativeOffset)`